src/Cli/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -201,77 +212,128 @@ private List<string> GetRestoreArguments()
 
             // --interactive need to output guide for auth. It cannot be
             // completely "quiet"
-            if (!RestoreArgs.Any(a => a.StartsWith("-verbosity:")))
+            if (!cliRestoreArgs.Any(a => a.StartsWith("-verbosity:")))
             {
                 var defaultVerbosity = Interactive ? "minimal" : "quiet";
                 args.Add($"-verbosity:{defaultVerbosity}");
             }
 
-            args.AddRange(RestoreArgs);
+            args.AddRange(cliRestoreArgs);
 
-            return args;
+            return args.ToArray();
         }
 
         private ICommand GetTargetCommand()
         {
-            var globalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
-            {
-                // This property disables default item globbing to improve performance
-                // This should be safe because we are not evaluating items, only properties
-                { Constants.EnableDefaultItems,    "false" },
-                { Constants.MSBuildExtensionsPath, AppContext.BaseDirectory }
-            };
+            // TODO for MSBuild usage here: need to sync loggers (primarily binlog) used with this evaluation
+            var project = EvaluateProject(ProjectFileFullPath, RestoreArgs);
+            InvokeRunArgumentsTarget(project);
+            var runProperties = ReadRunPropertiesFromProject(project, Args);
+            var command = CreateCommandFromRunProperties(project, runProperties);
+            return command;
 
-            if (!string.IsNullOrWhiteSpace(Configuration))
+            static ProjectInstance EvaluateProject(string projectFilePath, string[] restoreArgs)
             {
-                globalProperties.Add("Configuration", Configuration);
+                var globalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+                {
+                    // This property disables default item globbing to improve performance
+                    // This should be safe because we are not evaluating items, only properties
+                    { Constants.EnableDefaultItems,  "false" },
+                    { Constants.MSBuildExtensionsPath, AppContext.BaseDirectory }
+                };
+
+                var userPassedProperties = DeriveUserPassedProperties(restoreArgs);
+                if (userPassedProperties is not null)
+                {
+                    foreach (var (key, values) in userPassedProperties)
+                    {
+                        globalProperties[key] = string.Join(";", values);
+                    }
+                }
```

Comment:
Here's that major difference again - we now detect and forward any global properties to the run evaluation. Future work here should probably also pull the loggers (binlog, terminal logger, console logger) and any configuration to apply to the actual target _execution_ as well, so that we can have a consistent experience between build/restore and run argument evaluation.